awx/api/views/__init__.py

```diff
@@ -405,11 +405,23 @@ class InstanceInstanceGroupsList(InstanceGroupMembershipMixin, SubListCreateAtta
     parent_model = models.Instance
     relationship = 'rampart_groups'
 
+    def unattach_validate(self, request):
+        sub_id, res = super().unattach_validate(request)
+        if res:
+            return (sub_id, res)
+        parent = self.get_parent_object()
```

Comment:
I don't think there's any need to obtain `parent` at all. All of these queries will be duplicates too.

Reply:
This is why I suggested overriding `is_valid_relation`